commands/admin.js

```diff
@@ -0,0 +1,154 @@
+const { SlashCommandBuilder } = require('@discordjs/builders')
+const { Permissions, MessageActionRow, MessageButton } = require('discord.js')
+const { getGuild, saveGuild } = require('../helpers/guildData')
+const { foemp } = require('../helpers/foemp')
+const { BUTTON_STYLES, DEFAULT_TIMEOUT } = require('../helpers/constants')
+
+async function verifyAdmin (interaction) {
+  const user = interaction.member
+  const guild = await getGuild(interaction.guildId)
+
+  const isGuildAdmin = user.permissions.has(Permissions.FLAGS.ADMINISTRATOR)
+  const isBotAdmin = guild.admins.includes(user.id)
+
+  if (!isGuildAdmin && !isBotAdmin) {
+    await interaction.reply('https://gph.is/g/4w8PDNj')
+    return false
+  }
+  return true
+}
+
+async function listAdmins (interaction) {
+  const guild = await getGuild(interaction.guildId)
+
+  // Check existence of admins
+  if (guild.admins.length === 0) {
+    await interaction.reply('Er zijn nog geen bot admins aangeduid op deze server!')
+    return
+  }
+
+  // Build reply
+  const memberManager = interaction.guild.members
+  const admins = await memberManager.fetch({ user: guild.admins })
+  let response = 'De bot admins voor deze server zijn:'
+  for (const [, admin] of admins) { response += `\n${admin.user.username}` }
+  await interaction.reply(response)
+}
+
+async function addAdmin (interaction) {
```

Comment:
Produces this error when adding someone as admin without having the permission to do so and then crashes out. The bot replies with the "da gade gy ni bepalen"-gif, but still adds that id to the admins list.
Error:
```
{
  level: 'info',
  message: 'JSON data saved to file',
  timestamp: '2022-04-18T15:38:56.095Z'
}
{
  level: 'error',
  message: 'DiscordAPIError: Interaction has already been acknowledged.',
  timestamp: '2022-04-18T15:38:56.477Z',
  stack: 'Error: DiscordAPIError: Interaction has already been acknowledged.\n' +
    '    at Object.error (/home/mintage/dev/nerdlandbot_js/helpers/logger.js:43:7)\n' +
    '    at Client.<anonymous> (/home/mintage/dev/nerdlandbot_js/nerdlandbot.js:98:9)\n' +
    '    at processTicksAndRejections (node:internal/process/task_queues:96:5)'
}
```

Reply:
good catch